flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/async/AsyncWaitOperator.java

```diff
@@ -301,8 +302,17 @@ private void outputCompletedElement() {
             queue.emitCompletedElement(timestampedCollector);
             // if there are more completed elements, emit them with subsequent mails
             if (queue.hasCompletedElements()) {
-                mailboxExecutor.execute(
-                        this::outputCompletedElement, "AsyncWaitOperator#outputCompletedElement");
+                try {
+                    mailboxExecutor.execute(
+                            this::outputCompletedElement,
+                            "AsyncWaitOperator#outputCompletedElement");
+                } catch (RejectedExecutionException mailboxClosedException) {
+                    // This rejection is not a problem since this element will be processed one more
```

Comment:
Add a comment that we are assuming `RejectedExecutionException` can only happen if the task/operator is being cancelled/aborted and we thus it's safe to drop/ignore all of the pending records?